ui/pages/confirmations/hooks/alerts/useSignatureAlerts.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { SignatureRequestType } from '../../types/confirm';
+import { isSIWESignatureRequest } from '../../utils';
+import useCurrentConfirmation from '../useCurrentConfirmation';
+
+const useSignatureAlerts = (): Alert[] => {
+  const { currentConfirmation } = useCurrentConfirmation();
+  const t = useI18nContext();
+
+  const alerts = useMemo<Alert[]>(() => {
+    const isSIWE = isSIWESignatureRequest(
+      currentConfirmation as SignatureRequestType
+    );
+    if (!isSIWE || !currentConfirmation?.msgParams) {
+      return [];
+    }
+
+    const { msgParams } = currentConfirmation;
+    const isSIWEDomainValid = isValidSIWEOrigin(msgParams as WrappedSIWERequest);
+    if (isSIWEDomainValid) {
+      return [];
+    }
+
+    const alert = {
+      field: 'requestFrom',
+      key: 'requestFrom',
+      message: t('confirmAlertModalMessageDomainMismatch'),
+      reason: t('confirmAlertModalTitleSignIn'),
+      severity: Severity.Danger,
+    } as Alert;
+
+    return [alert];
+  }, [currentConfirmation?.id, currentConfirmation?.msgParams, t]);
+
+  return alerts;
+};
+
+export default useSignatureAlerts;
```

Comment:
Transaction specific hooks should be avoided I think, we added hook like `useBlockaidAlert` so that hook can be used for any transaction that we want.

Reply:
after discussion, we realized we want to rename the file and thus its purpose. Updated to `useDomainMismatchAlerts`. Thank you and lmk me know if you have any further suggestions